http/multipart/multipart.go

```diff
@@ -0,0 +1,57 @@
+package multipart
+
+import (
+	"errors"
+	"fmt"
+	ioutils "github.com/jfrog/gofrog/io"
+	"io"
+	"mime/multipart"
+	"os"
+)
+
+type FileWriterFunc func(fileName string) (io.WriteCloser, error)
+
+func ReadFilesFromStream(multipartReader *multipart.Reader, fileWriterFunc FileWriterFunc) error {
+	for {
+		// Read the next file streamed from client
+		fileReader, err := multipartReader.NextPart()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return fmt.Errorf("failed to read file: %w", err)
+		}
+		fileName := fileReader.FileName()
+		fileWriter, err := fileWriterFunc(fileName)
+		if err != nil {
+			return err
+		}
+
+		if _, err = io.Copy(fileWriter, fileReader); err != nil {
+			return fmt.Errorf("failed writing '%s' file: %w", fileName, err)
+		}
+		ioutils.Close(fileWriter, &err)
```

Comment:
Shouldn't we close it with a defer statement in case one of the errors occurs earlier? Perhaps we should move the loop logic into a separate function to ensure proper defer handling.

Reply:
I'm not sure @omerzi, see my comment https://github.com/jfrog/gofrog/pull/53/files#r1518785894

Reply:
@Or-Geva, if we move the logic inside the loop into a new function, the closing will occur on each iteration.